packages/eslint-plugin/src/rules/require-localize-metadata.ts

```diff
@@ -0,0 +1,100 @@
+import type { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [
+  {
+    readonly requireDescription?: boolean;
+  },
+];
+
+const DEFAULT_OPTIONS: Options[number] = {
+  requireDescription: false,
+};
+
+const VALID_LOCALIZED_STRING_WITH_DESCRIPTION = new RegExp(
+  /:(.*\|)?([\w\s]+){1}(@@.*)?:.+/,
+);
+
+// TODO: The following values are also used in eslint-plugin-template/i18n, and should likely be centralized
+const STYLE_GUIDE_LINK = 'https://angular.io/guide/i18n';
+const STYLE_GUIDE_LINK_COMMON_PREPARE = `${STYLE_GUIDE_LINK}-common-prepare`;
+const STYLE_GUIDE_LINK_METADATA_FOR_TRANSLATION = `${STYLE_GUIDE_LINK_COMMON_PREPARE}#i18n-metadata-for-translation`;
+
+export type MessageIds = 'requireLocalizeMetadata';
+export const RULE_NAME = 'require-localize-metadata';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that $localize tagged messages contain a description.',
+      recommended: false,
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          requireDescription: {
+            type: 'boolean',
+            default: DEFAULT_OPTIONS.requireDescription,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      requireLocalizeMetadata: `$localize tagged messages should contain a description. See more at ${STYLE_GUIDE_LINK_METADATA_FOR_TRANSLATION}`,
+    },
+  },
+  defaultOptions: [DEFAULT_OPTIONS],
+  create(context, [{ requireDescription }]) {
+    function reportRequireLocalizeMetadataError(
```

Comment:
Please can you refactor this implementation to just inline everything in the Node handler method?

There is very little code here but the dedication to wrapping every step into functions hurts readability in my opinion because of how it changes the flow of the document.

Reply:
See my comment below about eventually being able to support custom ID and meaning.

Reply:
Makes sense @adbaran1 👍